discovery-provider/src/queries/get_users_cnode.py

```diff
@@ -67,7 +72,10 @@ def get_users_cnode(cnode_endpoint_string, replica_type=ReplicaType.PRIMARY):
                     't.secondary1 = :cnode_endpoint_string OR '
                     't.secondary2 = :cnode_endpoint_string) AND'
             }
-            t.secondary1 is not NULL;
+            t.user_id > { prev_user_id } AND
+            t.secondary1 is not NULL
+            ORDER BY "user_id"
+            LIMIT { max_users };
```

Comment:
per slack discussion, lets make this back-compat so old CN can still fetch all users with no limit enforced and add comment for next release to enforce max limit

Reply:
we should also not do the `t.user_id > { prev_user_id } AND` clause if not specified
i think its fine since it defaults to 0 but that'll just make the query slower?
can add enforcement for that same as above

Reply:
good points -- fixed!